ra_aid/file_listing.py

```
@@ -95,42 +99,66 @@ def get_file_listing(
-
-        # Sort for consistency
-        files.sort()
-
-        # Get total count before truncation
-        total_count = len(files)
-
-        # Truncate if limit specified
+        # Get list of files from git ls-files
+        try:
+            # Get both tracked and untracked files
+            tracked_files_process = subprocess.run(
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)